realsense2_camera/src/base_realsense_node.cpp

```diff
@@ -401,6 +415,42 @@ bool BaseRealSenseNode::get_version_cb(realsense2_camera_srvs::srv::VersionReq::
     return true;
 }
 
+bool BaseRealSenseNode::get_pixel_cb(realsense2_camera_srvs::srv::PixelReq::Request::SharedPtr req, realsense2_camera_srvs::srv::PixelReq::Response::SharedPtr res)
+{
+    auto msg_camera_info = _camera_info[COLOR]; 
+    std::vector<geometry_msgs::msg::Point> pixels;
+    pixels.reserve(req->points_requested.size());
+    for(auto point: req->points_requested)
+    { 
+        auto transformed_point = point;
+        geometry_msgs::msg::Point pixel;
+        if(abs(point.point.x) + abs(point.point.y) + abs(point.point.z) > 0.1)
```

Comment:
Its checking if the point sent does not have coordinates 0.0, 0.0, 0.0. These coordinates would produce a NaN pixel as it would cause a zero division. I'm comparing this to 0.1 to avoid using == 0.0,  as they say its not a good practice to compare floats directly. The minimum range of the stereo is approximately 0.3m, so if the point is 0.0, 0.0, 0.0 the condition will be met

Reply:
Although the stereo range is >0.3m for example when using nav2 the path starts right after the chassis, so the point can be <0.3m but >0.

Reply:
A better check would be to check only that `x` __and__ `y` are greater than 0

Reply:
I see. So do you think this condition should be removed? or should I lower this condition to a very small value like 0.0001

Reply:
I changed this condition to the `z` coordinate of the transformed point being greater than zero, as this is the only element that causes this zero division. Will update the PR with this and the other required changes

Reply:
Agree with you! Better that way